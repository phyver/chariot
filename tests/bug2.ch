data list('x) where
    | Nil : list('x)
    | Cons : 'x -> list('x) -> list('x)

val test2 f [] x = []
  | test2 f x [] = []
  | test2 f (x::xs) (y::ys) = test2 f (y::xs) (x::ys)
