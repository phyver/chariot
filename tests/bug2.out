!! warning: the definition test is NOT provably correct (weight_bound: 2, depth_bound: 2)
